client/src/components/remixPaste.js

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+// how about instead /remix/:name renders a new component RemixPaste instead
+// RemixPaste checks the id when the page loads
+//and renders a textarea with the initial value of whatever the paste is
+
+// when the Remix Button is pressed
+// Navigates to the new component remixPaste.js
+// make context page that passes in the data, so youre able to pass it on
+// once it gets to the page, the text area is already open with the previous message inside
+// when the buttone is pressed the page renders to the new page ../{name}
+
+const RemixPaste = () => {
+  // const location = useLocation();
+
+  const [data, setData] = useState(null);
+  const [title, setTitle] = useState("");
+  const [remix, setRemix] = useState("Save");
+  const [message, setMessage] = useState("");
+  const [isEditing, setIsEditing] = useState(false);
+
+  // extracts name from the url
+  let { name } = useParams();
+
+  // http://localhost:3000/FormatFile/name
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        if (name) {
+          const res = await fetch(`http://localhost:8000/find/${name}`);
+          const file = await res.json();
+          setData(file);
+          setTitle(file.file[1]);
+          setMessage(file.show_file);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [title]);
+
+  if (!data) {
+    return <div>Loading...</div>;
+  }
+
+  const changeData = async (message) => {
+    try {
+      const res = await fetch("http://localhost:8000/store_paste", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ text: message }),
+      });
+
+      if (!res) {
+        throw new Error("No response saved");
```

Comment:
why would we throw an error? maybe instead we render something to the user? like have a hook for an error message, and set it to true

check out the example in this code on how we render errors to a user

https://github.com/SCE-Development/Clark/blob/d6b9ce481002ad72b0e339c2fbba5ab81649745f/src/Pages/Speaker/Speaker.js#L69

Reply:
Chnaged the error, ty

Reply:
Took out the hooks for save and remix button, works fine still